src/App.tsx

```diff
@@ -1,203 +1,93 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+
+import { Header } from './commponents/Header';
+import { TodoList } from './commponents/TodoList';
+import { Footer } from './commponents/Footer';
+
+enum Status {
+  Active = 'active',
+  Completed = 'completed',
+  All = 'all',
+}
+
+enum HasError {
+  UnableToLoad = 'Unable to load todos',
+  TitleNotEmpty = 'Title should not be empty',
+  UnableToAdd = 'Unable to add a todo',
+  UnableToDelete = 'Unable to delete a todo',
+  UnableToUpdate = 'Unable to update a todo',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
-
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
-
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
-          />
-
-          {/* Add a todo on form submit */}
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          {/* This is a completed todo */}
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                checked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Completed Todo
-            </span>
-
-            {/* Remove button appears only on hover */}
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtered, setFiltered] = useState<Status>(Status.All);
+  const [isError, setIsError] = useState<string>('');
 
-            {/* overlay will cover the todo while it is being deleted or updated */}
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+  useEffect(() => {
+    const fetchTodo = async () => {
+      try {
+        const isTodos = await getTodos();
```

Comment:
Try to make up better name
We start variable name with `is` only when it contains boolean value